css/src/components/checkbox.scss

```diff
@@ -0,0 +1,112 @@
+/* stylelint-disable selector-max-specificity, max-nesting-depth, selector-no-qualifying-type */
+
+$checkbox-border-radius: $radius-sm !default;
+$checkbox-unchecked-border-color: $text !default;
+$checkbox-color: $primary !default;
+$checkbox-check-color: $primary-invert !default;
+$checkbox-check-hover-color: $text-subtle !default;
+$checkbox-size: 1.25em !default;
+$checkbox-icon-size: 0.75em !default;
+$checkbox-timing-function: $input-timing-function !default;
+$checkbox-duration: $input-transition-duration !default;
+$checkbox-spacing: $spacer-3;
+
+@mixin checkmark {
+	display: block;
+	position: absolute;
+	top: 0.1875em;
+	width: 0.3125em;
+	height: $checkbox-icon-size;
+	transform: rotate(45deg);
+	border: solid $checkbox-check-hover-color;
+	border-width: 0 0.078em 0.078em 0;
+	content: '';
+}
+
+.checkbox {
+	display: flex;
+	position: relative;
+	align-items: center;
+	line-height: 1.25;
+	cursor: pointer;
+
+	.checkbox-check {
+		display: flex;
+		flex-shrink: 0;
+		align-items: center;
+		justify-content: center;
+		width: $checkbox-size;
+		height: $checkbox-size;
+		transition-duration: $checkbox-duration;
+		transition-property: background, border, border-color;
+		transition-timing-function: $checkbox-timing-function;
+		border: 0.078em solid $checkbox-unchecked-border-color;
+		border-radius: $checkbox-border-radius;
+
+		&:not(:last-child) {
+			margin-inline-end: 0.5rem;
+		}
+
+		&::before {
+			content: '\0020';
+		}
+	}
+
+	&:hover {
```

Comment:
nit: change to single selector, nesting doesn't provide any less duplication after recent changes  `.checkbox-check::before:hover`

Reply:
`.checkbox-check::before:hover` seems to lose the hover effect, but `.checkbox-check:hover::before` seems to work.

Reply:
Actually, that removed the checkmark on label hover. Updated to `&:hover .checkbox-check::before`